flake.nix

```diff
@@ -79,94 +58,25 @@
 
       perSystem = { config, system, pkgs, ... }:
         let
-          pkgs = import nixpkgs {
-            system = system;
-            overlays = [
-              rust-overlay.overlays.default
-            ];
-          };
-
-          inherit (pkgs) lib;
-
-          uefi-rust-stable = pkgs.rust-bin.fromRustupToolchainFile ./rust/stub/rust-toolchain.toml;
-          craneLib = crane.lib.x86_64-linux.overrideToolchain uefi-rust-stable;
-
-          # Build attributes for a Rust application.
-          buildRustApp = lib.makeOverridable (
-            { src
-            , target ? null
-            , doCheck ? true
-            , extraArgs ? { }
-            }:
-            let
-              commonArgs = {
-                inherit src;
-                CARGO_BUILD_TARGET = target;
-                inherit doCheck;
-
-                # Workaround for https://github.com/ipetkov/crane/issues/262.
-                dummyrs = pkgs.writeText "dummy.rs" ''
-                  #![allow(unused)]
-
-                  #![cfg_attr(
-                    any(target_os = "none", target_os = "uefi"),
-                    no_std,
-                    no_main,
-                  )]
-
-                  #[cfg_attr(any(target_os = "none", target_os = "uefi"), panic_handler)]
-                  fn panic(_info: &::core::panic::PanicInfo<'_>) -> ! {
-                      loop {}
-                  }
-
-                  #[cfg_attr(any(target_os = "none", target_os = "uefi"), export_name = "efi_main")]
-                  fn main() {}
-                '';
-              } // extraArgs;
-
-              cargoArtifacts = craneLib.buildDepsOnly commonArgs;
-            in
-            {
-              package = craneLib.buildPackage (commonArgs // {
-                inherit cargoArtifacts;
-              });
-
-              clippy = craneLib.cargoClippy (commonArgs // {
-                inherit cargoArtifacts;
-                cargoClippyExtraArgs = "-- --deny warnings";
-              });
-
-              rustfmt = craneLib.cargoFmt (commonArgs // { inherit cargoArtifacts; });
-            }
-          );
-
-          stubCrane = buildRustApp {
-            src = craneLib.cleanCargoSource ./rust/stub;
-            target = "x86_64-unknown-uefi";
-            doCheck = false;
-          };
-
-          fatStubCrane = stubCrane.override {
-            extraArgs = {
-              cargoExtraArgs = "--no-default-features --features fat";
+          pkgs = import nixpkgs { inherit system; };
+          uefiPkgs = import nixpkgs {
+            inherit system;
+            crossSystem = {
+              # linuxArch is wrong here, it will yield arm64 instead of aarch64.
+              config = "${pkgs.hostPlatform.qemuArch}-windows";
+              rustc.config = "${pkgs.hostPlatform.qemuArch}-unknown-uefi";
```

Comment:
```
              config = "${pkgs.stdenv.hostPlatform.qemuArch}-windows";
              rustc.config = "${pkgs.stdenv.hostPlatform.qemuArch}-unknown-uefi";
```


```
       error: attribute 'hostPlatform' missing

       at /nix/store/k7h8k8whqw64rmsg6yhkmmdc82lkczc6-source/flake.nix:66:27:

           65|               # linuxArch is wrong here, it will yield arm64 instead of aarch64.
           66|               config = "${pkgs.hostPlatform.qemuArch}-windows";
             |                           ^
           67|               rustc.config = "${pkgs.hostPlatform.qemuArch}-unknown-uefi";
       Did you mean rustPlatform?
```

Reply:
see https://github.com/nix-community/lanzaboote/pull/198